openedx/core/djangoapps/django_comment_common/comment_client/thread.py

```diff
@@ -193,27 +194,11 @@ def unFlagAbuse(self, user, voteable, removeAll):
         voteable._update_from_response(response)
 
     def pin(self, user, thread_id):
-        url = _url_for_pin_thread(thread_id)
-        params = {'user_id': user.id}
-        response = utils.perform_request(
-            'put',
-            url,
-            params,
-            metric_tags=self._metric_tags,
-            metric_action='thread.pin'
-        )
-        self._update_from_response(response)
+        thread_data = forum_api.pin_thread(user.id, thread_id)
+        self._update_from_response(thread_data)
```

Comment:
I wonder how we can get rid of these obnoxious calls to `_update_from_response`. That method blindly updates all attributes:

    def _update_from_response(self, response_data):
        for k, v in response_data.items():
            if k in self.accessible_fields:
                setattr(self, k, v)
            else:
                log.warning(
                    "Unexpected field {field_name} in model {model_name}".format(
                        field_name=k,
                        model_name=self.__class__.__name__
                    )
                )

If the API call accidentally returns a field that is in "accessible_fields", then that field will be updated, and there is no way that unit tests will catch that error.

Reply:
I think, if the API returns a value that is in self.accessible_fields, it should set it as it was before. As accessible_fields are already defined for each model, it will only set those or otherwise log.
May be I don't get the point.